src/RecipesBase.jl

```diff
@@ -74,20 +74,26 @@ function _equals_symbol(arg::Expr, sym::Symbol)
 end
 
 # build an apply_recipe function header from the recipe function header
-function get_function_def(func_signature::Expr)
-    # for parametric definitions, take the "curly" expression and add the func
+function get_function_def(func_signature::Expr, args::Vector)
     front = func_signature.args[1]
-    func = :(RecipesBase.apply_recipe)
-    if isa(front, Expr) && front.head == :curly
-        front.args[1] = func
-        func = front
+    if func_signature.head == :where
+        Expr(:where, get_function_def(front, args), [esc(arg) for arg in func_signature.args[2:end]]...)
```

Comment:
Yes, that's nicer. For some reason I had the impression that was not working, but I was probably having another syntax error. Thanks!